### PR TITLE
Bug/gsye 674 proposal2

### PR DESCRIPTION
**Pull request by hannesdiedrich**
## Reason for the proposed changes

Please describe what we want to achieve and why.

## Proposed changes

-

**INTEGRATION_TESTS_BRANCH**=master
**GSY_FRAMEWORK_BRANCH**=master

### CODE REVIEW

**Comment by hannesdiedrich** [comment]
@spyros The following integration test is failing and I think it has to be corrected:
`Simulation publishes intermediate and final results`
This is the step that fails is: 
```
@then('intermediate results are transmitted on every slot and final results once')
def interm_res_report(context):
    # Add an extra result for the start of the simulation.
    # Add another extra result for: initializing -> running
    assert context.interm_results_count == 14
```
I understand that `Add an extra result for the start of the simulation.` is triggered [here](https://github.com/gridsingularity/gsy-e/blob/a89cf46124942a45ede542cdbf63327263b43410/src/gsy_e/gsy_e_core/simulation/simulation.py#L162) .
And that `Add another extra result for: initializing -> running` is triggered [here](https://github.com/gridsingularity/gsy-e/blob/a89cf46124942a45ede542cdbf63327263b43410/src/gsy_e/gsy_e_core/simulation/simulation.py#L116).
However, in the integration test, the simulation is already inititated in a step before and the latter is not counting into the number of transmitted results, because the `simulation._results.kafka_connection.is_enabled()` is not mocked yet.
TLDR; The expected count should be 13 (1 for the start of the simulation and 12 for the 12 simulated market slots).
And this should be also in the current state where we send the results an additional time at the end of the simulation ([here](https://github.com/gridsingularity/gsy-e/blob/a89cf46124942a45ede542cdbf63327263b43410/src/gsy_e/gsy_e_core/simulation/simulation.py#L256)).

**Review by spyrostz — APPROVED** [review]
LGTM. I am not 100% convinced or the current_market -> spot_market rename, however the former name was also not clear therefore I do not have a strong opinion on it.

**Comment by spyrostz** [comment]
> @spyros The following integration test is failing and I think it has to be corrected: `Simulation publishes intermediate and final results` This is the step that fails is:
> 
> ```
> @then('intermediate results are transmitted on every slot and final results once')
> def interm_res_report(context):
>     # Add an extra result for the start of the simulation.
>     # Add another extra result for: initializing -> running
>     assert context.interm_results_count == 14
> ```
> 
> I understand that `Add an extra result for the start of the simulation.` is triggered [here](https://github.com/gridsingularity/gsy-e/blob/a89cf46124942a45ede542cdbf63327263b43410/src/gsy_e/gsy_e_core/simulation/simulation.py#L162) . And that `Add another extra result for: initializing -> running` is triggered [here](https://github.com/gridsingularity/gsy-e/blob/a89cf46124942a45ede542cdbf63327263b43410/src/gsy_e/gsy_e_core/simulation/simulation.py#L116). However, in the integration test, the simulation is already inititated in a step before and the latter is not counting into the number of transmitted results, because the `simulation._results.kafka_connection.is_enabled()` is not mocked yet. TLDR; The expected count should be 13 (1 for the start of the simulation and 12 for the 12 simulated market slots). And this should be also in the current state where we send the results an additional time at the end of the simulation ([here](https://github.com/gridsingularity/gsy-e/blob/a89cf46124942a45ede542cdbf63327263b43410/src/gsy_e/gsy_e_core/simulation/simulation.py#L256)).

I think that this was clarified and fixed with commit #a079584

**Comment by hannesdiedrich** [comment]
> LGTM. I am not 100% convinced or the current_market -> spot_market rename, however the former name was also not clear therefore I do not have a strong opinion on it.

Yeah, I know. I kept the old the naming towards the results and only renamed it internally.
